open-media-match/src/OpenMediaMatch/storage/default.py

```diff
@@ -149,21 +149,29 @@ def _get_bank(self, name: str) -> t.Optional[database.Bank]:
             select(database.Bank).where(database.Bank.name == name)
         ).scalar_one_or_none()
 
-    def bank_update(self, bank: BankConfig, *, create: bool = False) -> None:
+    def bank_update(
+        self,
+        bank: BankConfig,
+        *,
+        create: bool = False,
+        rename_from: t.Optional[str] = None,
+    ) -> None:
         if create:
             database.db.session.add(database.Bank.from_storage_iface_cls(bank))
         else:
-            database.db.session.execute(
-                update(database.Bank)
-                .where(database.Bank.name == bank.name)
-                .values(enabled_ratio=bank.matching_enabled_ratio)
-            )
+            previous = database.Bank.query.filter_by(
+                name=rename_from if rename_from is not None else bank.name
+            ).one_or_404()
+            previous.name = bank.name
+            previous.enabled_ratio = bank.matching_enabled_ratio
```

Comment:
I'm surprised this works - you don't have to make an explicit update?

Reply:
I was surprised as well, but it seems that because previous is a model instance, SQLAlchemy tracks changes you do on it and does the update automatically when you commit.

Reply:
also, previous code doing the .update().where().values() was somehow not triggering name validation, that's why I changed it to this format.

Reply:
Oh man, that's good to know, good catch